force-app/main/default/classes/EventTriggerHandler.cls

```diff
@@ -0,0 +1,62 @@
+public with sharing class EventTriggerHandler {
+    public static void preventOverlappingFutureEventsForAssignedUser(List<Event> triggerDotNew) {
+        Map<Id, List<Event>> ownerAssignedIdToEvents = createAssignedIdToEventListMap(triggerDotNew);
+        for(Event ev: triggerDotNew){
```

Comment:
Would it be better for the variable to be a little longer than one or two letters, I know the instructors weren't too keen on that one.